test/test_hyper.py

```diff
@@ -275,6 +276,18 @@ def test_closed_connections_are_reset(self):
         assert c.recent_stream is None
         assert c.next_stream_id == 1
         assert c.window_manager is not wm
+        with c._conn as conn:
+            assert conn.state_machine.state == ConnectionState.IDLE
+
+        c = HTTP20Connection('www.google.com')
```

Comment:
@Lukasa Yes, if we close a connection without `_sock` object before this patch. The state of that connection should be `CLOSED` and it would make assertion `conn.state_machine.state == ConnectionState.IDLE`  fail. On the other hand,  maybe we can check if the connection holds a different `H2Connection` instance to make it more clearly.